lib/apicast/proxy_rules_source.rb

```diff
@@ -0,0 +1,26 @@
+# frozen_string_literal: true
+
+module Apicast
+  class ProxyRulesSource
+    attr_reader :proxy
+
+    delegate :backend_api_configs, to: :proxy
+
+    def initialize(proxy)
+      @proxy = proxy
+    end
+
+    def to_hash
+      api_proxy_rules = []
+
+      backend_api_configs.each do |config|
+        config.backend_api.proxy_rules.each do |proxy_rule|
+          proxy_rule.pattern = "/#{config.path}#{proxy_rule.pattern}"
```

Comment:
`File.join`?

Reply:
This is kinda dangerous .... If there is another process that save the proxy_ruyle we might end up saving unwanted changes

How about decorate it? using config as second parameter of the Decorator?

Reply:
true, true.. apicast decorator could be the thing.. thanks!